### ISSUE TITLE
Check if `with pytest.raises()` CM block has exactly one statement inside

### ISSUE THREAD

**Issue by sobolevn** [m-burst/flake8-pytest-style#1]
# Rule request

I want a rule that prevents users from writing potentially incorrect tests.

## Thesis

This is what I have in mind.

Correct use:
```python
def test_xxx():
    tested_thing = do_some_init()
    with pytest.raises(SomeError):
        tested_thing.some_meth()  # Should raise SomeError
```

Potentially problematic use:
```python
def test_xxx():
    with pytest.raises(SomeError):
        tested_thing = do_some_init()  # <- even if the test worked properly when first comitted, there's a chance that this line will start raising SomeError at some point in the future making the next one unreachable, thus turning this test into an incorrect one
        tested_thing.some_meth()  # Should raise SomeError
```

## Reasoning

This would prevent silly mistakes in tests.

Original issue is opened by @webknjaz here: https://github.com/wemake-services/wemake-python-styleguide/issues/1110

**Comment by m-burst** [comment]
Hi @sobolevn @webknjaz,

Do you think we should also forbid control flow structures within `with pytest.raises()` blocks?

For example:
```py
def test_xxx():
    with pytest.raises(SomeError):
        if some_condition:
            do_smth()
            do_smth_else()
        else:
            do_smth_different()
```
Technically the CM block contains one `if` statement, the test behaviour is still unclear.

I would appreciate your opinion on this. Personally I'm leaning towards forbidding `if` and `with` statements, loops, and `try` blocks within `with pytest.raises()` CM blocks.

**Comment by sobolevn** [comment]
I agree: it should be just one expression. Not a complex node.

**Comment by m-burst** [comment]
Released v0.4.0 with rule `PT012` checking for this. /cc @sobolevn @webknjaz

**Comment by sobolevn** [comment]
Awesome! Thanks a lot!
